src/create-theme-provider.js

```diff
@@ -0,0 +1,92 @@
+const React = require('react');
+const PropTypes = require('prop-types');
+const isFunction = require('is-function');
```

Comment:
@philpl @kof I'm okay with this. One thing though is that delegating this check to separate package means delegating the tests as well. If we want to make this built-in then we need to write and maintain tests for that on our own. Would one of you want to step in and it?